### ISSUE TITLE
Container image does not deploy to an ACA environment (v0.11.127)

### ISSUE THREAD

**Issue by jarrydpage** [Azure/data-api-builder#2163]
### What happened?

The latest (`0.11.127`) container image does not successfully deploy to an Azure Container App environment, and errors with the following:

```json
{
  "status": "Failed",
  "error": {
    "code": "DeploymentFailed",
    "target": "/subscriptions/.../resourceGroups/dab-demo/providers/Microsoft.Resources/deployments/dab-on-aca",
    "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.",
    "details": [{
      "code": "InvalidParameterValueInContainerTemplate",
      "message": "The following field(s) are either invalid or missing. Field 'template.containers.dm-dab-aca-app.image' is invalid with details: 'Invalid value: \"mcr.microsoft.com/azure-databases/data-api-builder:0.11.127\": image OS/Arc must be linux/amd64 but found windows/amd64';."
    }]
  }
}
```

This can be replicated with the Azure [sample script](https://github.com/Azure/data-api-builder/blob/main/samples/azure/azure-container-apps-deploy.sh), `azure-container-apps-deploy.sh`.

Using version `0.10.23`, the application is able to successfully deploy without issues.

### Version

0.11.127

### What database are you using?

Azure SQL

### What hosting model are you using?

Container Apps

### Which API approach are you accessing DAB through?

REST

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by marcusrangell** [comment]
The version before, 0.11.126-rc, is the same. 0.11.12-rc is the latest one working on Linux.

**Comment by jodyford-msft** [comment]
I get the same error on 0.11.12-rc.  But, I also have to update this line: 
Microsoft.App/containerApps@2022-10-01 
to this 
Microsoft.App/containerApps@2022-11-01-preview

**Comment by jarrydpage** [comment]
Looks like it was triggered by the change in the `Dockerfile` under dbcb6515f1dec4aa7f21e69e2c1f245cab739400

```bash
$ curl -s "https://mcr.microsoft.com/v2/azure-databases/data-api-builder/manifests/0.10.21" | jq -r '.history[0].v1Compatibility' | jq '.os'
"linux"
$ curl -s "https://mcr.microsoft.com/v2/azure-databases/data-api-builder/manifests/0.11.12-rc" | jq -r '.history[0].v1Compatibility' | jq '.os'
"linux"
$ curl -s "https://mcr.microsoft.com/v2/azure-databases/data-api-builder/manifests/latest" | jq -r '.history[0].v1Compatibility' | jq '.os'
"windows"
$ curl -s "https://mcr.microsoft.com/v2/azure-databases/data-api-builder/manifests/0.11.126-rc" | jq -r '.history[0].v1Compatibility' | jq '.os'
"windows"
```

**Comment by abhishekkumams** [comment]
the latest release uses an windows image, due to which this issue is occurring.

**Comment by GregDSmith** [comment]
> the latest release uses an windows image, due to which this issue is occurring.

I hope this was not intention? As noted above this has broken our Azure Container App environment deployments because they are Linux based

**Comment by jodyford-msft** [comment]
yeah, I had a demo today...

**Comment by abhishekkumams** [comment]
As a workaround, https://github.com/Azure/data-api-builder/blob/main/Dockerfile
this file can be used to build the linux based docker image locally. In the meantime, we are working on pushing the linux based image

**Comment by abhishekkumams** [comment]
we just released v0.11.130, we have updated it back to use linux image

**Comment by jarrydpage** [comment]
<img width="225" alt="image" src="https://github.com/Azure/data-api-builder/assets/1924697/75967c3e-df87-4fc3-a4b3-ff05cc20029e">


I can confirm the latest image works fine now, thanks!
